generators/csharp/sdk/src/SdkGeneratorContext.ts

```diff
@@ -277,16 +277,21 @@ export class SdkGeneratorContext extends AbstractCsharpGeneratorContext<SdkCusto
         return `${upperFirst(camelCase(this.config.organization))}${this.ir.apiName.pascalCase.unsafeName}`;
     }
 
-    public getRootClientClassName(): string {
+    public getRootClientClassName({ asSnippet }: { asSnippet?: boolean } = {}): string {
+        if (asSnippet) {
+            if (this.customConfig["exported-client-class-name"] != null) {
+                return this.customConfig["exported-client-class-name"];
+            }
+        }
         if (this.customConfig["client-class-name"] != null) {
             return this.customConfig["client-class-name"];
         }
         return `${this.getComputedClientName()}Client`;
     }
 
-    public getRootClientClassReference(): csharp.ClassReference {
+    public getRootClientClassReference({ asSnippet }: { asSnippet?: boolean } = {}): csharp.ClassReference {
         return csharp.classReference({
-            name: this.getRootClientClassName(),
+            name: this.getRootClientClassName({ asSnippet }),
             namespace: this.getNamespace()
         });
     }
```

Comment:
Having a separate function may be cleaner than passing this `asSnippet` argument. It'll have more duplicated code, but maybe cleaner.

Reply:
I actually started with that approach, then refactored to this afterwards. Given that you called it out, I'll refactor it back 👍